docs/14-connectors.md

```diff
@@ -0,0 +1,380 @@
+  src: 'hero/image-small.jpg'
+  alt: 'alt text'
+```
+
+As you can see this is the pattern that our preprocessor needs to return. The data.yaml file is used to mock variables and texts,
+so one Muban is compiled and you access Muban preview, you exactly see these texts.
+What we gonna do now, we will create a preprocessor that takes data from paragraph fields and returns it in same format as in provided data.yaml file.
+
++ First of all we need to create following folders inside of our custom module 'mymodule_components':
++ src/Plugin/ThemeEntityProcessor/[EntityType] - EntityType can be Paragraph, Node, Media etc... 
+
+In our case it will be a preprocessor for paragraph src/Plugin/ThemeEntityProcessor/ParagraphsBlock
+Inside of this folder, we will create a PHP Class that will match the name of our c01_hero component.
+
++ src/Plugin/ThemeEntityProcessor/ParagraphC01Hero.php
+
+and the contents of file will look like this: 
+```
+<?php
+
+namespace Drupal\c01_hero\Plugin\ThemeEntityProcessor\Paragraph;
+
+use Drupal\handlebars_theme_handler\Plugin\ThemeEntityProcessorBase;
+
+/**
+ * Returns the structured data of an entity.
+ *
+ * @ThemeEntityProcessor(
+ *   id = "c01_hero",
+ *   label = @Translation("C01 Hero"),
+ *   entity_type = "paragraph",
+ *   bundle = "c01_hero",
+ *   view_mode = "default"
+ * )
+ */
+class ParagraphC17ContactInfoLinkItem extends ThemeEntityProcessorBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function preprocessItemData(&$variables) {
+    $image = $this->themeFieldProcessorManager->getFieldData($variables['elements']['field_image']);
+    $title = $this->themeFieldProcessorManager->getFieldData($variables['elements']['field_title']);
+    $body = $this->themeFieldProcessorManager->getFieldData($variables['elements']['field_body']);
+    $variables['data'] = [
+      'title' => $title,
+      'body' => $body,
+      'image' => [
+        'src' => $image['image']['src'], 
+        'alt' => $image['image']['alt']
+        ]
+    ];
+  }
+}
+
+```
+
+Now let's take a look at the annotation of the class, as you can see it contains:
++ id - this is an id of the preprocessor, usually we keep it as [machineName]_[viewMode], it should be unique.
++ label - human readable label
++ entity_type - in our case is paragraph
++ bundle - is machine name of the entity, in our case c01_hero
++ view_mode - the view mode for which we create preprocessor, in our case is default. In future you can create more view modes for entities and create a separate preprocessor per view mode.
+
+The annotation is automatically decoded by ThemeEntityProcessor and used further.
+Also you can note that the main class is matching the file name and it extends the ThemeEntityProcessorBase, this is a requirement.
+Is not excluded that you can create an intermediate class MyModuleIntermediateBase extends ThemeEntitiyProcessorBase in case you prefer to change something in the middle, here you use the power of OOP and Dependency Injection.
```

Comment:
```suggestion
In case you need more flexibility, you can create an intermediate class `MyModuleIntermediateBase extends ThemeEntitiyProcessorBase` so you can change something in the middle. Use the power of OOP and Dependency Injection to your advantage.
```